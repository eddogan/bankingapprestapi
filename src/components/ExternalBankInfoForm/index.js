import React, { Fragment, useState, useEffect, useCallback } from "react";
import axios from "axios";
// Material-UI imports
import { Grid, Typography, Box, TextField } from "@material-ui/core/";
import { makeStyles } from "@material-ui/core/styles";
// Custom imports
import getApi from "../../helpers/GetApi";

export default function ExternalBankInfoForm() {
  const classes = useStyles();
  const [bankInfo, setBankInfo] = useState({
    bankName: "",
    routingNumber: "",
    accountNumber: "",
    accountType: ""
  });

  const fetchBank = useCallback(async () => {
    const clientBankAccountEndpoint = `${process.env.REACT_APP_API_URL}/api/clients/${sessionStorage.AccountId}/bankaccounts`;
    const clientBankAccountResponse = await axios(
      getApi(clientBankAccountEndpoint)
    );
    // Mask account number
    let accountNumber = clientBankAccountResponse.data[0].accountNumber;
    accountNumber = `************${accountNumber.substring(
      accountNumber.length - 4
    )}`;
    clientBankAccountResponse.data[0].accountNumber = accountNumber;

    setBankInfo(clientBankAccountResponse.data[0]);
  }, []);

  useEffect(() => {
    fetchBank();
  }, [fetchBank]);

  return (
    <Fragment>
      <Box mb={2}>
        <div className="moduleContent">
          <form>
            <Grid container justify="space-between" alignItems="center">
              <Grid item xs={12} sm={12} md={12} lg={12}>
                <Box px={2}>
                  <Typography
                    className={classes.moduleSubTitle}
                    variant="h6"
                    component="h1"
                  >
                    External Bank Information
                  </Typography>
                </Box>
              </Grid>
              <Grid item xs={12} sm={12} md={12} lg={12}>
                <Box p={2}>
                  <Typography variant="body2">
                    Changes to External Bank Information may be made by calling
                    Customer Care at
                  </Typography>
                </Box>
              </Grid>
              <Grid item xs={12} sm={12} md={6} lg={6}>
                <Box px={2}>
                  <TextField
                    InputLabelProps={{ shrink: true }}
                    id="bankName"
                    name="bankName"
                    label="Bank name"
                    variant="outlined"
                    margin="normal"
                    fullWidth={true}
                    value={bankInfo.bankName}
                    disabled
                  />
                </Box>
              </Grid>
              <Grid item xs={12} sm={12} md={6} lg={6}>
                <Box px={2}>
                  <TextField
                    InputLabelProps={{ shrink: true }}
                    id="bankRoutingNumber"
                    name="bankRoutingNumber"
                    label="Bank routing number"
                    variant="outlined"
                    margin="normal"
                    fullWidth={true}
                    value={bankInfo.routingNumber}
                    disabled
                  />
                </Box>
              </Grid>
              <Grid item xs={12} sm={12} md={6} lg={6}>
                <Box px={2}>
                  <TextField
                    InputLabelProps={{ shrink: true }}
                    id="bankAccountNumber"
                    name="bankAccountNumber"
                    label="Bank account number"
                    variant="outlined"
                    margin="normal"
                    fullWidth={true}
                    value={bankInfo.accountNumber}
                    disabled
                  />
                </Box>
              </Grid>
              <Grid item xs={12} sm={12} md={6} lg={6}>
                <Box px={2}>
                  <TextField
                    InputLabelProps={{ shrink: true }}
                    id="bankAccountType"
                    name="bankAccountType"
                    label="Bank account type"
                    variant="outlined"
                    margin="normal"
                    fullWidth={true}
                    value={bankInfo.accountType}
                    disabled
                  />
                </Box>
              </Grid>
            </Grid>
          </form>
        </div>
      </Box>
    </Fragment>
  );
}

const useStyles = makeStyles(theme => ({
  moduleSubTitle: {
    borderBottom: "1px dashed #eee"
  }
}));
