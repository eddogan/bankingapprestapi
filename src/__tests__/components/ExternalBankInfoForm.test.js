import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import ExternalBankInfoForm from "../../components/ExternalBankInfoForm";

it("renders correctly", () => {
  const tree = renderer.create(<ExternalBankInfoForm />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<ExternalBankInfoForm />, div);
  ReactDOM.unmountComponentAtNode(div);
});
