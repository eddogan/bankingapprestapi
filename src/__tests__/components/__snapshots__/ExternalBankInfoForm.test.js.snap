// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-3"
>
  <div
    className="moduleContent"
  >
    <form>
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-12"
        >
          <div
            className="MuiBox-root MuiBox-root-107"
          >
            <h1
              className="MuiTypography-root makeStyles-moduleSubTitle-1 MuiTypography-h6"
            >
              External Bank Information
            </h1>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-12"
        >
          <div
            className="MuiBox-root MuiBox-root-138"
          >
            <p
              className="MuiTypography-root MuiTypography-body2"
            >
              Changes to External Bank Information may be made by calling Customer Care at
            </p>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-6"
        >
          <div
            className="MuiBox-root MuiBox-root-139"
          >
            <div
              className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-disabled Mui-disabled"
                data-shrink={true}
                htmlFor="bankName"
              >
                Bank name
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root Mui-disabled Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl"
                onClick={[Function]}
              >
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-197 MuiOutlinedInput-notchedOutline"
                  style={
                    Object {
                      "paddingLeft": 8,
                    }
                  }
                >
                  <legend
                    className="PrivateNotchedOutline-legend-198"
                    style={
                      Object {
                        "width": 0,
                      }
                    }
                  >
                    <span
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "&#8203;",
                        }
                      }
                    />
                  </legend>
                </fieldset>
                <input
                  aria-invalid={false}
                  className="MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-disabled"
                  disabled={true}
                  id="bankName"
                  name="bankName"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-6"
        >
          <div
            className="MuiBox-root MuiBox-root-199"
          >
            <div
              className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-disabled Mui-disabled"
                data-shrink={true}
                htmlFor="bankRoutingNumber"
              >
                Bank routing number
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root Mui-disabled Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl"
                onClick={[Function]}
              >
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-197 MuiOutlinedInput-notchedOutline"
                  style={
                    Object {
                      "paddingLeft": 8,
                    }
                  }
                >
                  <legend
                    className="PrivateNotchedOutline-legend-198"
                    style={
                      Object {
                        "width": 0,
                      }
                    }
                  >
                    <span
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "&#8203;",
                        }
                      }
                    />
                  </legend>
                </fieldset>
                <input
                  aria-invalid={false}
                  className="MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-disabled"
                  disabled={true}
                  id="bankRoutingNumber"
                  name="bankRoutingNumber"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-6"
        >
          <div
            className="MuiBox-root MuiBox-root-200"
          >
            <div
              className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-disabled Mui-disabled"
                data-shrink={true}
                htmlFor="bankAccountNumber"
              >
                Bank account number
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root Mui-disabled Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl"
                onClick={[Function]}
              >
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-197 MuiOutlinedInput-notchedOutline"
                  style={
                    Object {
                      "paddingLeft": 8,
                    }
                  }
                >
                  <legend
                    className="PrivateNotchedOutline-legend-198"
                    style={
                      Object {
                        "width": 0,
                      }
                    }
                  >
                    <span
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "&#8203;",
                        }
                      }
                    />
                  </legend>
                </fieldset>
                <input
                  aria-invalid={false}
                  className="MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-disabled"
                  disabled={true}
                  id="bankAccountNumber"
                  name="bankAccountNumber"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-6"
        >
          <div
            className="MuiBox-root MuiBox-root-201"
          >
            <div
              className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-disabled Mui-disabled"
                data-shrink={true}
                htmlFor="bankAccountType"
              >
                Bank account type
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root Mui-disabled Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl"
                onClick={[Function]}
              >
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-197 MuiOutlinedInput-notchedOutline"
                  style={
                    Object {
                      "paddingLeft": 8,
                    }
                  }
                >
                  <legend
                    className="PrivateNotchedOutline-legend-198"
                    style={
                      Object {
                        "width": 0,
                      }
                    }
                  >
                    <span
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "&#8203;",
                        }
                      }
                    />
                  </legend>
                </fieldset>
                <input
                  aria-invalid={false}
                  className="MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-disabled"
                  disabled={true}
                  id="bankAccountType"
                  name="bankAccountType"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
